controller/handler/trial.js

```diff
@@ -32,161 +32,159 @@ let processCurrentTrial = '';
-            const ruleValues = rules.map((ruleData) => {
-                return parseInt(ruleData.rule, 10);
-            });
+        );
 
-            complianceCount = processComplianceCount(compliance);
-            patientCount = xPatients.length;
-            const patientStatuses = compliance.map(processPatientStatus);
-
-            patientArray = xPatients.map((patient) => {
-                const patientStatus = patientStatuses.find((status) => {
-                    return status.PatientPin === patient.PatientPin;
-                });
-
-                let missedWeekly = missedLastWeek.find((missed) => {
-                    return missed.PatientPinFK === patient.PatientPin;
-                });
-
-                if (missedWeekly) {
-                    if (typeof patient.lastWeekly === 'undefined') {
-                        if (missedWeekly.State === 'expired') {
-                            patient.lastWeekly = 'Missed';
-                        } else if (missedWeekly.State === 'completed') {
-                            patient.lastWeekly = 'Taken';
-                        }
-                    }
-                } else {
-                    patient.lastWeekly = ' ---- ';
-                }
+        const rules = [];
 
-                // collect the compliance status as well as expiredCount
-                if (patientStatus) {
-                    patient.trialStatus = patientStatus.trialStatus;
-                    patient.status = patientStatus.status;
-                    if (isNaN(patientStatus.compliancePercentage)) {
-                        patient.compliancePercentage = ' ---- ';
-                    } else {
-                        patient.compliancePercentage = patientStatus.compliancePercentage;
-                    }
-                    if (isNaN(patientStatus.trendingCompliance)) {
-                        patient.trendingCompliance = ' ---- ';
-                    } else {
-                        patient.trendingCompliance = patientStatus.trendingCompliance;
-                    }
-                } else {
-                    patient.status = 'Pending';
-                }
-                patient.DateStarted = moment.utc(patient.DateStarted)
-                    .format('MM-DD-YYYY');
-                patient.DateCompleted = moment.utc(patient.DateCompleted)
-                    .format('MM-DD-YYYY');
+        if (!xCurrentTrial) {
+            throw new Error('trial does not exist');
+        }
+        const ruleValues = rules.map((ruleData) => {
+            return parseInt(ruleData.rule, 10);
+        });
 
-                return patient;
+        complianceCount = processComplianceCount(compliance);
+        patientCount = xPatients.length;
+        const patientStatuses = compliance.map(processPatientStatus);
+
+        patientArray = xPatients.map((patient) => {
+            const patientStatus = patientStatuses.find((status) => {
+                return status.PatientPin === patient.PatientPin;
```

Comment:
:art: arrow functions don't need a block if it returns the first element.

```js
patientStatuses.find(status => status.PatientPin === patient.PatientPin);
```

Reply:
I tried to to do as you suggested but lint is not allowing this format with "=>" function. In lint it is required to have block statements surrounded by arrow body with return statement and parentheses around arrow function argument. Do we need to change some rule or something like that for the same ?

Reply:
https://eslint.org/docs/rules/arrow-body-style `as-needed` option

Reply:
Done. Thanks :)